pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/SegmentDeletionManager.java

```diff
@@ -140,7 +140,11 @@ private synchronized void deleteSegmentFromPropertyStoreAndLocal(String tableNam
         if (fileToMove.exists()) {
           File targetDir = new File(new File(_localDiskDir, DELETED_SEGMENTS), rawTableName);
           try {
-            FileUtils.moveFileToDirectory(fileToMove, targetDir, true);
+            // Overwrites the file if it already exists in the target directory.
+            FileUtils.copyFileToDirectory(fileToMove, targetDir, true);
+            if (!fileToMove.delete()) {
```

Comment:
can you re-order the log statement. Print the move statement first and then try the delete. Also if failing to delete must be warn/error ?